Cabal/src/Distribution/Simple/Build/PathsModule/Z.hs

```diff
@@ -85,11 +85,7 @@ render z_root = execWriter $ do
   then do
     tell "#if defined(VERSION_base)\n"
     tell "\n"
-    tell "#if MIN_VERSION_base(4,0,0)\n"
     tell "catchIO :: IO a -> (Exception.IOException -> IO a) -> IO a\n"
-    tell "#else\n"
-    tell "catchIO :: IO a -> (Exception.Exception -> IO a) -> IO a\n"
-    tell "#endif\n"
```

Comment:
This must be restored. While we don't support building Cabal itself with old GHCs, iirc we do support building packages with new Cabal and old GHCs. Paths module generation affects the latter.